core/comment_events.js

```diff
@@ -155,21 +177,40 @@ Blockly.Events.CommentChange.prototype.isNull = function() {
  * @param {boolean} forward True if run forward, false if run backward (undo).
  */
 Blockly.Events.CommentChange.prototype.run = function(forward) {
-  var workspace = this.getEventWorkspace_();
-  var comment = workspace.getCommentById(this.commentId);
+  var comment = this.getComment_();
   if (!comment) {
     console.warn('Can\'t change non-existent comment: ' + this.commentId);
     return;
   }
   var contents = forward ? this.newContents_ : this.oldContents_;
 
-  comment.setContent(contents);
+  if (goog.isString(contents)) {
```

Comment:
Sounds good. I’ll just make it an object instead of a string or object. I was considering doing that anyways but originally thought this might be a less invasive change.

In terms of not being able to have the multiple state items change in a single change event, I made it that way because a single user interaction shouldn’t be able to change both, and we’re never emitting a CommentChange event with multiple state changes. I figure if we ever need that, I’d rather the developer have to explicitly change this piece of code for this new kind of functionality.

Reply:
Re my previous comment in this thread, I ended up making the change to allow multiple types of comment changes at once (e.g. minimized state and text).